### PR TITLE
gh-121272: move __future__ import validation from compiler to symtable

### PR DESCRIPTION
**Pull request by iritkatriel**
* Issue: gh-121272

### CODE REVIEW

**Review by AraHaan — APPROVED** [review]
I agree, this could be a good performance improvement when using a ton of c extensions with python scripts that use import from statements on them.

File: Python/symtable.c [hunk 1]
```diff
@@ -1914,6 +1935,8 @@ symtable_visit_stmt(struct symtable *st, stmt_ty s)
         break;
     case ImportFrom_kind:
         VISIT_SEQ(st, alias, s->v.ImportFrom.names);
+        if (!check_import_from(st, s))
+            VISIT_QUIT(st, 0);
```
Comment (picnixz) [inline Python/symtable.c]:
```suggestion
        if (!check_import_from(st, s)) {
            VISIT_QUIT(st, 0);
        }
```

Not sure if PEP 7 applies here as "new code".

**Comment by iritkatriel** [comment]
> I agree, this could be a good performance improvement when using a ton of c extensions with python scripts that use import from statements on them.

This shouldn't impact performance.

This is about simplifying code in compile.c (in this case, making ``compiler_from_import`` no longer access the internals of ``struct compiler``). This is part of a longer term plan to break up the compiler.

**Review by JelleZijlstra — APPROVED** [review]
Looks good. This is thoroughly tested in `test_future_stmt/test_future.py` and I confirmed that the code in future.c that sets `ff_location` makes sense.

**Pull request merged**